src/spyglass/utils/mixins/export.py

```diff
@@ -0,0 +1,333 @@
+        )
+        self._export_table.File.insert(
+            [{"export_id": self.export_id, tbl_pk: fname} for fname in fnames],
+            skip_duplicates=True,
+        )
+        fnames_str = "('" + "', ".join(fnames) + "')"  # log AnalysisFile table
+        table()._log_fetch(restriction=f"{tbl_pk} in {fnames_str}")
+
+    def _run_join(self, **kwargs):
+        """Log join for export.
+
+        Special case to log primary keys of each table in join, avoiding
+        long restriction strings.
+        """
+        table_list = [self]
+        other = kwargs.get("other")
+
+        if hasattr(other, "_log_fetch"):  # Check if other has mixin
+            table_list.append(other)  # can other._log_fetch
+        else:
+            logger.warning(f"Cannot export log join for\n{other}")
+
+        joined = self.proj().join(other.proj(), log_export=False)
+        for table in table_list:  # log separate for unique pks
+            for r in joined.fetch(*table.primary_key, as_dict=True):
+                table._log_fetch(restriction=r)
+
+    def _run_with_log(self, method, *args, log_export=True, **kwargs):
+        """Run method, log fetch, and return result.
+
+        Uses FETCH_LOG_FLAG to prevent multiple logs in one user call.
+        """
+        log_this_call = FETCH_LOG_FLAG.get()  # One log per fetch call
+
+        if log_this_call and not self.database == "common_usage":
+            FETCH_LOG_FLAG.set(False)
+
+        try:
+            ret = method(*args, **kwargs)
+        finally:
+            if log_this_call:
+                FETCH_LOG_FLAG.set(True)
+
+        if log_export and self.export_id and log_this_call:
+            if getattr(method, "__name__", None) == "join":  # special case
+                self._run_join(**kwargs)
+            else:
+                self._log_fetch(restriction=kwargs.get("restriction"))
+            logger.debug(f"Export: {self._called_funcs()}")
+
+        return ret
+
+    # -------------------------- Intercept DJ methods --------------------------
+
+    def fetch(self, *args, log_export=True, **kwargs):
+        """Log fetch for export."""
+        if not self.export_id:
+            return super().fetch(*args, **kwargs)
+        return self._run_with_log(
+            super().fetch, *args, log_export=log_export, **kwargs
+        )
+
+    def fetch1(self, *args, log_export=True, **kwargs):
+        """Log fetch1 for export."""
+        if not self.export_id:
+            return super().fetch1(*args, **kwargs)
+        return self._run_with_log(
+            super().fetch1, *args, log_export=log_export, **kwargs
+        )
+
+    def restrict(self, restriction):
+        """Log restrict for export."""
+        if not self.export_id:
+            return super().restrict(restriction)
+        log_export = "fetch_nwb" not in self._called_funcs()
+        return self._run_with_log(
+            super().restrict, restriction=restriction, log_export=log_export
```

Comment:
To help keep from catching too many things with sequential & restrictions, could we do:
```suggestion
            super().restrict, restriction=dj.AndList(restriction, self.restriction, log_export=log_export
```

Reply:
would also need to import `AndList` from `datajoint.condition` at the top of the file

Reply:
Sure - this would reduce our `Table & r1 & r2` problem from `Table & r1 OR Table & r2` to `Table & r1 OR Table & r1 & r2`, eliminating `r2 and not r1` items. It makes explaining the over-selection more complicated though. If the problem persists, I have a slight preference for an easier message of 'AND -> OR', but I could be swayed. 

@edeno - what do you think?

Reply:
It is more technical to explain the whole thing, but best practices for a user during their code development would then be:

- when doing a series of restrictions, put the most restrictive key first
- when in doubt, AndList

Running the export with this saved me some rewriting of code to use AndList as a n=1 user case

Reply:
I do think 'AND -> OR' is easier to explain, but will still be quite difficult for people to understand in practice. I think practically even asking people to follow the best practices above is going to be tricky.